lang/LangPrimSource/PyrPrimitive.cpp

```diff
@@ -438,31 +443,73 @@ int prObjectString(struct VMGlobals* g, int numArgsPushed) {
     }
 }
 
+void trimTrailingZeros(char* str) {
+    char* point = strrchr(str, '.');
+    if (point != nullptr) {
+        char* end = str + strlen(str) - 1;
+        while (end > point && *end == '0') {
+            *end-- = '\0';
+        }
+        if (end == point) { // If only the decimal point is left, remove it too
+            *end = '\0';
+        }
+    }
+}
+
 int prFloat_AsStringPrec(struct VMGlobals* g, int numArgsPushed);
 int prFloat_AsStringPrec(struct VMGlobals* g, int numArgsPushed) {
     PyrSlot* a = g->sp - 1;
     PyrSlot* b = g->sp;
 
     int precision;
     int err = slotIntVal(b, &precision);
-    if (err)
-        return err;
+    if (err) {
+        return err; // Return error if precision cannot be obtained
+    }
+
+    double value = slotRawFloat(a);
+
+    if (std::isnan(value) || std::isinf(value)) {
+        return errFailed;
+    }
+
+    // Calculate the number of significant digits in the value
+    // This approach assumes normal floating-point numbers and avoids edge cases
+    int significantDigits = value == 0 ? 1 : std::max(1, static_cast<int>(std::log10(std::fabs(value)) + 1));
```

Comment:
I tried to improve, but this is still better. The only way is just to avoid getting significant digits at all.

https://github.com/supercollider/supercollider/pull/6228#issuecomment-1977139506

Reply:
Yeah, in the overall context it's probably fine. It's not exactly a performance critical function.